### DISCUSSION TITLE
While using permission middleware it throws error

### DISCUSSION THREAD

**Discussion by yasirmehmood123** [spatie/laravel-permission#1953] [General]
I was trying to use the permissions for a role and everything was added as per documentation like adding middleware in kernel.php
let me give an example how I am using it
`Route::group(['middleware' => ['can:view customers']], function () {`

Let me show the error as well
**Call to a member function contains() on array**

File directory where I found it
"\vendor\spatie\laravel-permission\src\Traits\HasPermissions.php:288"
I fixed it by replacing line number 288 of HasPermission trait let me show an example of this as well
`return $this->permissions->contains($permission->getKeyName(), $permission->getKey());`
to
`return collect($this->permissions)->contains($permission->getKeyName(), $permission->getKey());`

please if i am wrong correct me but if this is a real issue than please update it in the coming updates

**Comment by erikn69** [comment]
`$this->permissions` is a relation `BelongsToMany` , so, somewhere on your implementation you are changing that collection to array, and that makes the error 
Read this [must-not-have-a-permission-or-permissions-property-nor-a-permissions-method](https://spatie.be/docs/laravel-permission/v5/prerequisites#must-not-have-a-permission-or-permissions-property-nor-a-permissions-method)
https://github.com/spatie/laravel-permission/blob/08cc9bdbf89640cb1d64fb01eb9513c1cbb2549d/src/Traits/HasPermissions.php#L45-L53
Also can is tested [GateTest](https://github.com/spatie/laravel-permission/blob/main/tests/GateTest.php)

**Reply by yasirmehmood123** [reply]
@erikn69 thanks a lot for the help.
You were right at one point i was converting it to array

**Reply by erikn69** [reply]
Mark as answered
